### ISSUE TITLE
Use different config files and do different PRs on one repository

### ISSUE THREAD

**Issue by teolemon** [crowdin/github-action#89]
### What
- We have a repository with PR averaging 1K to 2K files, and I'd like to split it up in smaller PRs (either one PR per language, or one PR per file to translate, not completely sure).
- Is there a way to use different config files and do different PRs on one repository, without spawning the GitHub action several times, and wasting computing ressources both on GitHub and on Crowdin's server ?

**Comment by andrii-bodnar** [comment]
HI @teolemon!

It is possible to use different config files and do different PRs on one repository.

To do this, you need to split your Crowdin Action WF step into separate steps. Each of these steps should use a different config through the `config` parameter. Also, you can specify various `pull_request_title`'s for each of these steps.

And then, each WF Step will use only a specified config file and will create a separate PR including only those files, that were configured in config.

**Comment by teolemon** [comment]
@andrii-bodnar Thanks a lot for the answer 👍  Am I right in thinking that this method will trigger a Crowdin build each time (around 45 min for us) ?
45 min * 100 = a lot of time
Ideally, I'd like to do it this way:
```
steps:
  - name: Common Build (45min) using the Crowdin action
strategy:
 matrix:
  lang: ['aa', 'fr', 'en', ...]
steps:
  - name: Language PR (10s) without having to trigger the action again
    run: echo ${{ matrix.lang }}
```

**Comment by andrii-bodnar** [comment]
@teolemon it's interesting :thinking: 

Just trying to figure out if this is possible now. Crowdin Action doesn't know anything about languages and the only way to control languages - parameters `download_language` and `upload_language`.

Let's imagine that we will download all the translations once and then we should create a separate PR for each language. Crowdin Action will need to find files for each language, then add to git and commit. This is a difficult task because downloaded files can have different export patterns with different language codes.

You could try to split your WF at least by language. In This case, Crowdin Action will create a separate PR for each language and as a result, PRs will be smaller. This shouldn't affect the execution time, because *project build* ≈ *lang1 build* + *lang2 build* + *...* + *langN build*

**Comment by teolemon** [comment]
Open Food Fact's current workflow is this one, https://github.com/openfoodfacts/openfoodfacts-server/blob/main/.github/workflows/crowdin.yml, with files located in another file https://github.com/openfoodfacts/openfoodfacts-server/blob/main/crowdin.yml

I've made a PR implementing your suggestion, but I'm not sure it won't block the 2nd step to the build triggered in the 1st one
https://github.com/openfoodfacts/openfoodfacts-server/pull/5752

**Comment by andrii-bodnar** [comment]
Hi @teolemon!
Hmm, unfortunately in this case all of the builds were running in parallel and some of them failed with *429 Too Many Requests* and *409 Conflict* responses. 

Parallel builds are not allowed in Crowdin and number of simultaneous API requests per account is 20 requests.

Is there any way to run a matrix not in parallel, but sequentially?

**Comment by teolemon** [comment]
There is @andrii-bodnar : https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymax-parallel

**Comment by andrii-bodnar** [comment]
@teolemon great, probably with `max-parallel: 1` it should work.

**Comment by andrii-bodnar** [comment]
https://github.com/crowdin/github-action/wiki/Separate-PRs-for-each-language
